extension/content/enginesView.mjs

```diff
@@ -162,6 +164,51 @@ export default class EnginesView extends HTMLElement {
       privateDefault ? privateDefault : "Unset";
   }
 
+  /**
+   * Sorts the search engines in a similar manner to
+   * SearchService._sortEnginesByDefaults
+   * https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/toolkit/components/search/SearchService.sys.mjs#2760
+   *
+   * @param {object[]} engines
+   *   An array of engines to be sorted in place.
+   * @param {object} privateDefault
+   *   The default engine object.
+   * @param {string} locale
+   *   The BCP47 locale for the user.
+   */
+  #sortEngines(engines, privateDefault, locale) {
+    // We pass the locale to ensure we use the same locale for comparison as
+    // desktop will in its comparison.
+    const collator = new Intl.Collator(locale);
+    const defaultEngine = engines[0];
+    engines.sort((a, b) => {
+      if (a == defaultEngine) {
+        return -1;
+      }
```

Comment:
I suggest adding the same comment as the Desktop code: 
> The app default engine should always be first in the list (except
    for distros, that we should respect).